From 1b5de04e1ae401f2f3d7179da0379191886cdfad Mon Sep 17 00:00:00 2001
From: Torge Matthies <tmatthies@codeweavers.com>
Date: Tue, 2 May 2023 01:36:12 +0200
Subject: [PATCH] libs/libjpeg: Set default DCT algorithm to fastest.

---
 libs/jpeg/jconfig.h | 1 +
 1 file changed, 1 insertion(+)

diff --git a/libs/jpeg/jconfig.h b/libs/jpeg/jconfig.h
index 2d05a3b09026..9f18c71751bd 100644
--- a/libs/jpeg/jconfig.h
+++ b/libs/jpeg/jconfig.h
@@ -17,6 +17,7 @@
 /* #undef NEED_SHORT_EXTERNAL_NAMES */
 /* Define this if you get warnings about undefined structures. */
 /* #undef INCOMPLETE_TYPES_BROKEN */
+#define JDCT_DEFAULT JDCT_FASTEST
 
 /* Define "boolean" as unsigned char, not enum, on Windows systems. */
 #ifdef _WIN32
-- 
2.40.1

