--- b/include/mmddk.h
+++ a/include/mmddk.h
@@ -30,6 +30,14 @@
 extern "C" {
 #endif
 
+#define MAX_MIDIINDRV 	(16)
+/* For now I'm making 16 the maximum number of midi devices one can
+ * have. This should be more than enough for everybody. But as a purist,
+ * I intend to make it unbounded in the future, as soon as I figure
+ * a good way to do so.
+ */
+#define MAX_MIDIOUTDRV 	(16)
+
 /* ==================================
  *   Multimedia DDK compatible part
  * ================================== */
