diff --git a/src/main.rs b/src/main.rs
index ba9b0dd..00df815 100644
--- a/src/main.rs
+++ b/src/main.rs
@@ -395,7 +395,7 @@ fn main() {
     );
     if let Ok(appid) = std::env::var("ROFI_INFO") {
         let _ = daemon(true, false);
-        let mut cmd = std::process::Command::new("xdg-open")
+        let mut cmd = std::process::Command::new("@xdg_open@")
             .arg(&format!("steam://rungameid/{appid}"))
             .spawn()
             .unwrap();
