From 2bc4fabd214939f24c7f980e80fe5f3f0c6ff687 Mon Sep 17 00:00:00 2001
From: chayleaf <chayleaf-git@pavluk.org>
Date: Fri, 23 Aug 2024 05:35:57 +0700
Subject: [PATCH 3/4] mobile: swaybar: bottom -> overlay layer

---
 swaybar/bar.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/swaybar/bar.c b/swaybar/bar.c
index 021fc3bd..7f44799d 100644
--- a/swaybar/bar.c
+++ b/swaybar/bar.c
@@ -103,7 +103,7 @@ static void add_layer_surface(struct swaybar_output *output) {
 	output->layer_surface = zwlr_layer_shell_v1_get_layer_surface(
 			bar->layer_shell, output->surface, output->output,
 			hidden || overlay ? ZWLR_LAYER_SHELL_V1_LAYER_OVERLAY :
-			ZWLR_LAYER_SHELL_V1_LAYER_BOTTOM, "panel");
+			ZWLR_LAYER_SHELL_V1_LAYER_OVERLAY, "panel");
 	assert(output->layer_surface);
 	zwlr_layer_surface_v1_add_listener(output->layer_surface,
 			&layer_surface_listener, output);
-- 
2.45.2

